pkg/apis/core/validation/validation.go

```diff
@@ -4196,12 +4199,19 @@ func ValidateNodeUpdate(node, oldNode *core.Node) field.ErrorList {
 		addresses[address] = true
 	}
 
-	if len(oldNode.Spec.PodCIDR) == 0 {
+	if len(oldNode.Spec.PodCIDRs) == 0 {
 		// Allow the controller manager to assign a CIDR to a node if it doesn't have one.
-		oldNode.Spec.PodCIDR = node.Spec.PodCIDR
+		//this is a no op for a string slice.
+		oldNode.Spec.PodCIDRs = node.Spec.PodCIDRs
 	} else {
-		if oldNode.Spec.PodCIDR != node.Spec.PodCIDR {
-			allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "podCIDR"), "node updates may not change podCIDR except from \"\" to valid"))
+		// compare the entire slice
+		if len(oldNode.Spec.PodCIDRs) != len(node.Spec.PodCIDRs) {
```

Comment:
So this was okay previously where we assume only 1 CIDR. If we keep this validation logic, it means that we can't add CIDRs in multiple patches/updates. Is that expected?

Reply:
Yes. cidr allocation (even in multi cidr case) is always done in a patch. CNI like kubenet does not really know how to work with multi `event` triggering `cidr change`.. they pick the first and ignore the rest..